backend/dal/Helpers/Extensions/RowVersionExtensions.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using Pims.Dal.Entities;
+
+namespace Pims.Dal.Helpers.Extensions
+{
+    /// <summary>
+    /// RowVersionExtensions static class, provides extension methods for RowVersions.
+    /// </summary>
+    public static class RowVersionExtensions
+    {
+        /// <summary>
+        /// Convert the string rowversion to a byte array.
+        /// </summary>
+        /// <param name="rowversion"></param>
+        /// <returns></returns>
+        public static byte[] ConvertRowVersion(this string rowversion)
+        {
+            return Convert.FromBase64String(rowversion);
+        }
+
+        /// <summary>
+        /// Convert the byte array rowversion to a string.
+        /// </summary>
+        /// <param name="rowversion"></param>
+        /// <returns></returns>
+        public static string ConvertRowVersion(this byte[] rowversion)
+        {
+            return Convert.ToBase64String(rowversion);
+        }
+
+        /// <summary>
+        /// Extract the rowversion from the entity and convert to a string.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public static string GetRowVersion(this BaseEntity entity)
+        {
+            return Convert.ToBase64String(entity.RowVersion);
+        }
+
+        /// <summary>
+        /// Copy the rowversion into the destination entity so that EF concurrency will work.
+        /// </summary>
+        /// <param name="dest"></param>
+        /// /// <param name="rowversion"></param>
+        public static void SetRowVersion(this BaseEntity dest, string rowversion)
+        {
+            var rv = Convert.FromBase64String(rowversion);
+            Buffer.BlockCopy(rv, 0, dest.RowVersion, 0, rv.Length);
+        }
+
+        /// <summary>
+        /// Copy the rowversion from the source entity to the destination entity so that EF concurrency will work.
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="dest"></param>
+        public static void CopyRowVersionTo(this BaseEntity source, BaseEntity dest)
```

Comment:
Consider adding logic to create an empty RowVersion array if the dest rowversion is null

Reply:
done